api-spec/extensions/query.fragment.yaml

```diff
@@ -0,0 +1,74 @@
+paths:
+  /stac/search:
+    get:
+      parameters:
+        - $ref: '#/components/parameters/query'
+components:
+  parameters:
+    query:
+      name: query
+      in: query
+      description: query for properties in items. Use the JSON for of the queryFilter used in POST.
+      required: false
+      schema:
+        type: string
+  schemas:
+    searchBody:
+      allOf:
+        - $ref: '#/components/schemas/queryFilter'
+    queryFilter:
+      type: object
+      description: Allows users to query properties for specific values
+      properties:
+        query:
+          $ref: '#/components/schemas/query'
+    query:
+      type: object
+      description: Define which properties to query and the operatations to apply
+      additionalProperties:
+        $ref: '#/components/schemas/queryProp'
+      example:
+        eo:cloud_cover:
+          lt: 50
+        provider: Planet
+        published:
+          date: '2018-02-12T00:00:00Z/2018-03-18T12:31:12Z'
+        pl:item_type:
+          startsWith: PSScene
+    queryProp:
+      description: Apply query operations to a specific property
+      anyOf:
+        - type: object
+          description: if the object doesn't contain any of the operators, it is equivalent to using the equals operator
+        - type: object
+          description: Match using an operator
+          properties:
+            eq:
+              type: object
+              description: Find items with a property that is equal to the specified value
+            neq:
+              type: object
+              description: Find items that *don't* contain the specified value
+            gt: 
+              type: number
+              description: Find items with a property value greater than the specified value
+            lt:
+              type: number
+              description: Find items with a property value less than the specified value
+            gte:
+              type: number
+              description: Find items with a property value greater than or equal the specified value
+            lte:
+              type: number
+              description: Find items with a property value greater than or equal the specified value
+            startsWith:
```

Comment:
Are startsWith, endsWith and contains case-sensitive or insensitive?

Reply:
I think case insensitive search gives better results. You never really know whether people will write "landsat" or "Landsat" or "LANDSAT". Saw all of them in API implementations for openEO and we would only get one result with case-sensitive. Sure, case insensitive is more work to be implemented. Let's make an issue out of it, then people can discuss.

Reply:
I sounded in on the issue - I think case insensitive makes a bit more sense, but don't feel strongly.

Reply:
I think I'm OK with stating case insensitive. This is intended to be a simplified search, so I'm not sure it makes sense to separate items by case